lib/ansible/module_utils/urls.py

```diff
@@ -1727,7 +1727,7 @@ def url_argument_spec():
 
 
 def fetch_url(module, url, data=None, headers=None, method=None,
-              use_proxy=True, force=False, last_mod_time=None, timeout=10,
+              use_proxy=None, force=False, last_mod_time=None, timeout=10,
```

Comment:
The fact that `use_proxy` is an explicit kwarg, means that `module` is not the right place to get the arg from.

What needs to happen, is that the `uri` module needs to explicitly pass `module.params['use_proxy']` to `fetch_url` instead.

We've effectively stopped using the ability to just use `module.params` directly here, and have instead been making them explicit kwargs.

Reply:
uri module still passes this explicitly, this was added due to my comment above (which is really reflecting the reason this issue stalled in previous incarnations). The change really only covers 'other modules' that include the option but don't pass it in.

Reply:
yep, as mentioned in the last comment 

> What needs to happen, is that the uri module needs to explicitly pass module.params['use_proxy'] to fetch_url instead.

this change it's included in my PR. @sivel 

https://github.com/ansible/ansible/blob/dcd9fe8bccfd42698792d79a135f2d0a231d51b2/lib/ansible/modules/uri.py#L595